lib/pharos/phases/configure_master.rb

```diff
@@ -70,6 +75,9 @@ def install
 
         copy_kube_certs
 
+        @ssh.exec!("sudo mkdir -p #{SECRETS_CFG_DIR}")
+        configure_secrets_encryption
```

Comment:
The `generate_config` gets used for both `install` and also `upgrade`, so the upgrade path is currently broken, leading to apiserver errors:

```
I0417 07:56:33.343388       1 server.go:121] Version: v1.9.6
error opening encryption provider configuration file "/etc/pharos/secrets-encryption/config.yml": open /etc/pharos/secrets-encryption/config.yml: no such file or directory
```

Depending on how kube itself handles migration to the encryption provider, this could work by also conditionally generating the `.../secrets-encryption/config.yml` as part of the `upgrade`?

Reply:
Kube itself does not do any migration automatically. It is doable with `kubectl` by "re-writing" the secrets with new encryption config in place.

I'll see if there's some way to make it work in upgrade too.

Reply:
Is it able to read both encrypted and existing unencrypted secrets when configured with `--experimental-encryption-provider-config=`?

I don't think we necessarily need any explicit migration to rewrite the secrets, but it would be good to handle configuration changes like this correctly on upgrades. The same comment about breaking the configuration on upgrades seems to also apply to the auth/audit webhooks, except those are `cluster.yml` changes which we don't support (yet?), whereas this is a new configuration that gets enabled by default.

Reply:
AFAIK it's able to read both encrypted and existing unencrypted secrets, needs maybe `identity` provider also configured as the last provider:

> When reading resources from storage each provider that matches the stored data attempts to decrypt the data in order.